README.md

```diff
@@ -72,27 +74,32 @@ worktree root.
 cleared so that you don't accidentally go backward to a different branch and
 edit the wrong files.
 
+`autopush`: When creating a new worktree, it will push the branch to the upstream then perform a `git rebase`
+
 ```lua
 require("git-worktree").setup({
     update_on_change = <boolean> -- default: true,
     clearjumps_on_change = <boolean> -- default: true,
+    autopush = <boolean> -- default: false,
 })
 ```
 
 ## Usage<a name="usage"></a>
 
 Three primary functions should cover your day-to-day.
 
+The path can be either relative from the git root dir or absoulut path to the worktree.
+
 ```lua
--- Creates a worktree.  Requires the branch name and the upstream
+-- Creates a worktree.  Requires the path, branch name, and the upstream
 -- Example:
-:lua require("git-worktree").create_worktree("feat-69", "upstream/master")
+:lua require("git-worktree").create_worktree("feat-69", "master", "origin")
 
--- switches to an existing worktree.  Requires the branch name
+-- switches to an existing worktree.  Requires the path name
```

Comment:
should we say branch instead of path?

Reply:
Right now we use relative or absolute path to indenting the worktree, built since you can only have 1 worktree per branch we could use that too

Do you prefer to use branch name ?

I figure most will just use the telescope plugin which displays both to filter upon

Thoughts?

Reply:
hmm.  Its a good point.

What is the proper way to handle this?  The hard part is that my branch name = path name.

Feels weird to have a world where that doesn't exist.  

It seems to make sense to consider branch as the thing you switch to.  but question for you, can it handle both?  Meaning we check for the provided item to be a branch, if its not, is it a path?  I think uv_real_path (used in plenary) should ensure that what is provided is a real path.

Reply:
when you use the git cli to remove it asks for the path to the worktree (also it lists the path first when displaying the worktree) so i think we need to support paths, but it should be doable to accept branch name too. 

either way works for your usecase since path == branch and telescope has both on the list to fuzzy find on